future-apps/payroll/contracts/Payroll.sol

```diff
@@ -137,192 +139,145 @@ contract Payroll is EtherTokenConstant, IForwarder, IsContract, AragonApp {
-     * @param _accountAddress Employee's address to receive payroll
-     * @param _initialDenominationSalary Employee's salary, per second in denomination token
-     * @param _name Employee's name
-     * @param _role Employee's role
-     * @param _startDate Employee's starting date (it actually sets their initial lastPayroll value)
-     */
-    function addEmployee(
-        address _accountAddress,
-        uint256 _initialDenominationSalary,
-        string _name,
-        string _role,
-        uint64 _startDate
-    )
+        _addEmployee(_accountAddress, _initialDenominationSalary, _name, _role, getTimestamp64());
+    }
+
+    /**
+     * @notice Adds employee `_name` with address `_accountAddress` to Payroll with a salary of `_initialDenominationSalary` per second, starting on `_startDate`.
+     * @param _accountAddress Employee's address to receive payroll.
+     * @param _initialDenominationSalary Employee's salary, per second in denomination token.
+     * @param _name Employee's name.
+     * @param _role Employee's role.
+     * @param _startDate Employee's starting timestamp in seconds (it actually sets their initial lastPayroll value).
+     */
+    function addEmployee(address _accountAddress, uint256 _initialDenominationSalary, string _name, string _role, uint64 _startDate)
         external
         authP(ADD_EMPLOYEE_ROLE, arr(_accountAddress, _initialDenominationSalary, _startDate))
     {
-        _addEmployee(
-            _accountAddress,
-            _initialDenominationSalary,
-            _name,
-            _role,
-            _startDate
-        );
+        _addEmployee(_accountAddress, _initialDenominationSalary, _name, _role, _startDate);
     }
 
     /**
-     * @notice Set employee #`_employeeId`'s annual salary to `_denominationSalary` per second.
-     * @param _employeeId Employee's identifier
-     * @param _denominationSalary Employee's new salary, per second in denomination token
+     * @notice Sets employee #`_employeeId`'s annual salary to `_denominationSalary` per second.
+     * @param _employeeId Employee's identifier.
+     * @param _denominationSalary Employee's new salary, per second in denomination token.
      */
-    function setEmployeeSalary(
-        uint256 _employeeId,
-        uint256 _denominationSalary
-    )
+    function setEmployeeSalary(uint256 _employeeId, uint256 _denominationSalary)
         external
-        employeeActive(_employeeId)
         authP(SET_EMPLOYEE_SALARY_ROLE, arr(_employeeId, _denominationSalary))
+        employeeActive(_employeeId)
     {
-        uint64 timestamp = getTimestamp64();
-
         // Add owed salary to employee's accrued value
-        uint256 owed = _getOwedSalary(_employeeId, timestamp);
+        uint256 owed = _getOwedSalary(_employeeId);
         _addAccruedValue(_employeeId, owed);
```

Comment:
⚠️ This is the only bit that's not nice with changing accrued value to "reimbursements". Changing your salary should not make your previous accrued salary a reimbursement.

A way around this would be to add another storage slot, `employee.accruedSalary`, and keep tracked of accrued salary there instead of adjusting the last payroll date on partial payrolls.

Reply:
Then I would just keep the old name ;-)

Reply:
The idea of using the word "reimburse" was to name the action of cashing out your accrued value, no te rename accrued value, somehow an analogy of `payday` to `salary`. Therefore, avoiding the word "reimburse" won't hide us from the fact that we are mixing accounting concepts here.

Reply:
I just realized bonuses would probably fall under accrued value as well, which would certainly not fall under reimbursements in the real world.

Wondering if it'd be too much to add another storage slot to keep track of these bonuses separately, just so we can use a different reference string. We could potentially use something like an enum argument to differentiate between the type of payment we'd like to do so the interface is streamlined into a single `payday()` and `partialPayday()` function.

I think salary + reimbursement + bonus _should_ cover all employee types of remuneration (anything related to a stock / token, e.g. options or dividends, would be a separate contract).

Reply:
Moving this one to a separate issue